source/bdDetect.py

```diff
@@ -210,6 +225,24 @@ class _DeviceInfoFetcher(AutoPropertyObject):
 	"""Utility class that caches fetched info for available devices for the duration of one core pump cycle."""
 	cachePropertiesByDefault = True
 
+	def __init__(self):
+		self._btDevsLock = threading.Lock
+		self._btDevsCache: typing.Optional[typing.Tuple[str, DeviceMatch]] = None
+
+	#: Type info for auto property: _get_btDevsCache
+	btDevsCache: typing.Optional[typing.List[typing.Tuple[str, DeviceMatch]]]

```

Comment:
it might be worth extracting this to a named type variable e.g. `btDevsCacheT = typing.Optional[typing.List[typing.Tuple[str, DeviceMatch]]]`